### ISSUE TITLE
💅 False positive for noUselessEscapeInRegex with named backreference

### ISSUE THREAD

**Issue by geoffswift** [biomejs/biome#6331]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.12.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              false
  Recommended:                  false
  All:                          true
  Enabled rules:
  suspicious/noCatchAssign
  complexity/useLiteralKeys
  complexity/noMultipleSpacesInRegularExpressionLiterals
  suspicious/useNamespaceKeyword
  complexity/noUselessEmptyExport
  nursery/useConsistentMemberAccessibility
  suspicious/noAssignInExpressions
  suspicious/noDuplicateSelectorsKeyframeBlock
  suspicious/noDuplicateParameters
  correctness/noConstructorReturn
  style/useSelfClosingElements
  correctness/noUnusedLabels
  correctness/noUnreachableSuper
  suspicious/noCompareNegZero
  a11y/noAutofocus
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noControlCharactersInRegex
  style/noVar
  suspicious/noDoubleEquals
  suspicious/noEmptyInterface
  suspicious/noConstEnum
  nursery/noDuplicateElseIf
  correctness/noPrecisionLoss
  nursery/noRestrictedImports
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noImplicitAnyLet
  a11y/useKeyWithClickEvents
  suspicious/noDuplicateObjectKeys
  suspicious/noUnsafeDeclarationMerging
  suspicious/noReactSpecificProps
  correctness/noInnerDeclarations
  correctness/useArrayLiterals
  nursery/useGoogleFontDisplay
  correctness/noInvalidDirectionInLinearGradient
  nursery/noValueAtRule
  nursery/noMissingVarFunction
  suspicious/noImportantInKeyframe
  complexity/noUselessLabel
  complexity/noUselessCatch
  a11y/useAriaPropsForRole
  nursery/noOctalEscape
  correctness/noNonoctalDecimalEscape
  style/noDoneCallback
  nursery/noDocumentImportInPage
  complexity/noUselessUndefinedInitialization
  nursery/noNestedTernary
  a11y/noInteractiveElementToNoninteractiveRole
  suspicious/noLabelVar
  correctness/noUnnecessaryContinue
  suspicious/noApproximativeNumericConstant
  nursery/noDuplicateCustomProperties
  correctness/noUnknownUnit
  correctness/noEmptyCharacterClassInRegex
  suspicious/noSparseArray
  a11y/useIframeTitle
  a11y/noSvgWithoutTitle
  correctness/noVoidElementsWithChildren
  correctness/useJsxKeyInIterable
  style/useExportType
  complexity/noUselessLoneBlockStatements
  style/noArguments
  a11y/useValidAriaValues
  suspicious/noGlobalAssign
  suspicious/noCommentText
  suspicious/noThenProperty
  suspicious/useGetterReturn
  style/useFragmentSyntax
  a11y/noPositiveTabindex
  nursery/noDuplicateProperties
  correctness/noRenderReturnValue
  correctness/useExhaustiveDependencies
  security/noGlobalEval
  style/noYodaExpression
  a11y/noRedundantRoles
  correctness/noUnusedVariables
  suspicious/noSelfCompare
  suspicious/noAsyncPromiseExecutor
  nursery/noDescendingSpecificity
  nursery/noIrregularWhitespace
  security/noDangerouslySetInnerHtml
  style/useNodejsImportProtocol
  suspicious/noArrayIndexKey
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  performance/noAccumulatingSpread
  suspicious/noConfusingLabels
  correctness/noChildrenProp
  correctness/noUnknownFunction
  correctness/noInvalidPositionAtImportRule
  nursery/noDocumentCookie
  nursery/useDeprecatedReason
  style/noShoutyConstants
  a11y/noAriaUnsupportedElements
  correctness/noFlatMapIdentity
  a11y/noBlankTarget
  a11y/useHeadingContent
  nursery/noDynamicNamespaceImportAccess
  correctness/useValidForDirection
  correctness/noInvalidUseBeforeDeclaration
  a11y/noAriaHiddenOnFocusable
  a11y/useGenericFontNames
  style/useBlockStatements
  correctness/noInvalidNewBuiltin
  style/noNegationElse
  style/noRestrictedGlobals
  style/useExponentiationOperator
  correctness/noUnusedFunctionParameters
  nursery/useSortedClasses
  style/noUnusedTemplateLiteral
  nursery/noStaticElementInteractions
  suspicious/useAwait
  nursery/useAriaPropsSupportedByRole
  nursery/useAdjacentOverloadSignatures
  style/noNamespace
  suspicious/noDuplicateAtImportRules
  complexity/noUselessFragments
  complexity/noUselessStringConcat
  correctness/noUnusedImports
  suspicious/noEmptyBlock
  suspicious/noFunctionAssign
  suspicious/noUnsafeNegation
  a11y/useValidLang
  suspicious/useValidTypeof
  a11y/useValidAriaRole
  correctness/noConstantCondition
  a11y/useAriaActivedescendantWithTabindex
  style/useExplicitLengthCheck
  style/useDefaultParameterLast
  complexity/noEmptyTypeParameters
  nursery/useComponentExportOnlyModules
  correctness/noUnknownProperty
  nursery/useTrimStartEnd
  complexity/noUselessTernary
  suspicious/noExplicitAny
  correctness/noSwitchDeclarations
  complexity/noUselessTypeConstraint
  suspicious/noRedundantUseStrict
  style/useLiteralEnumMembers
  suspicious/noGlobalIsNan
  suspicious/noSkippedTests
  suspicious/noMisleadingCharacterClass
  a11y/noLabelWithoutControl
  suspicious/noRedeclare
  correctness/noStringCaseMismatch
  nursery/noUselessStringRaw
  nursery/noImgElement
  complexity/noUselessThisAlias
  correctness/noUnreachable
  suspicious/noFallthroughSwitchClause
  complexity/noThisInStatic
  suspicious/noDuplicateCase
  a11y/useValidAnchor
  complexity/useRegexLiterals
  correctness/noSelfAssign
  correctness/noInvalidBuiltinInstantiation
  nursery/useGuardForIn
  style/useShorthandFunctionType
  suspicious/noShadowRestrictedNames
  style/useThrowNewError
  a11y/useMediaCaption
  nursery/useAtIndex
  correctness/noUnsafeFinally
  style/useNodeAssertStrict
  nursery/noHeadElement
  nursery/useExplicitType
  style/useEnumInitializers
  a11y/useHtmlLang
  nursery/noRestrictedTypes
  suspicious/noEvolvingTypes
  style/useWhile
  complexity/useArrowFunction
  style/noInferrableTypes
  a11y/noNoninteractiveTabindex
  complexity/useSimpleNumberKeys
  correctness/useYield
  style/useNumericLiterals
  suspicious/noImportAssign
  suspicious/useDefaultSwitchClauseLast
  nursery/noDuplicatedFields
  correctness/noGlobalObjectCalls
  style/noParameterProperties
  a11y/useAltText
  suspicious/noSuspiciousSemicolonInJsx
  complexity/noBannedTypes
  style/useThrowOnlyError
  suspicious/noPrototypeBuiltins
  style/useAsConstAssertion
  nursery/noEnum
  suspicious/noDebugger
  suspicious/noMisleadingInstantiator
  complexity/noVoid
  nursery/noUnknownPseudoClass
  a11y/useFocusableInteractive
  correctness/noUnmatchableAnbSelector
  suspicious/noDuplicateJsxProps
  correctness/noEmptyPattern
  performance/noReExportAll
  security/noDangerouslySetInnerHtmlWithChildren
  a11y/useKeyWithMouseEvents
  suspicious/noExtraNonNullAssertion
  suspicious/noShorthandPropertyOverrides
  nursery/noUnknownPseudoElement
  style/useConst
  style/noNonNullAssertion
  complexity/useFlatMap
  correctness/useHookAtTopLevel
  correctness/useIsNan
  suspicious/noGlobalIsFinite
  suspicious/noDuplicateFontNames
  complexity/noExcessiveCognitiveComplexity
  a11y/noDistractingElements
  nursery/useValidAutocomplete
  style/useConsistentArrayType
  style/useForOf
  nursery/useStrictMode
  a11y/useValidAriaProps
  a11y/noRedundantAlt
  complexity/useDateNow
  nursery/noHeadImportInDocument
  suspicious/noConfusingVoidType
  suspicious/noFocusedTests
  a11y/useButtonType
  a11y/useSemanticElements
  correctness/noInvalidGridAreas
  nursery/noUnknownTypeSelector
  style/useShorthandArrayType
  style/useShorthandAssign
  suspicious/useErrorMessage
  correctness/noConstantMathMinMaxClamp
  correctness/noUnusedPrivateClassMembers
  correctness/noVoidTypeReturn
  correctness/noUnknownMediaFeatureName
  nursery/noUselessEscapeInRegex
  a11y/useAnchorContent
  complexity/noUselessRename
  nursery/noSubstr
  style/useNumberNamespace
  complexity/noUselessConstructor
  a11y/noAccessKey
  suspicious/noExportsInTest
  suspicious/useNumberToFixedDigitsArgument
  a11y/noNoninteractiveElementToInteractiveRole
  nursery/noTemplateCurlyInString
  style/noCommaOperator
  suspicious/useIsArray
  nursery/useConsistentCurlyBraces
  correctness/noNewSymbol
  a11y/noHeaderScope
  suspicious/noMisrefactoredShorthandAssign
  suspicious/noClassAssign
  style/noImplicitBoolean
  style/useImportType

Workspace:
  Open Documents:               0
```

### Rule name

noUselessEscapeInRegex

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=LwAoAD8APABxAHUAbwB0AGUAPgAnAHwAIgApACgALgAqACkAXABrADwAcQB1AG8AdABlAD4ALwA%3D

### Expected result

`\k` is the appropriate representation of a named backreference, and should not be flagged as a useless escape sequence.  Refer here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_backreference

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
